openmc/model/surface_composite.py

```diff
@@ -52,6 +53,92 @@ def __neg__(self):
         """Return the negative half-space of the composite surface."""
 
 
+class CylinderSector(CompositeSurface):
+    """Infinite cylindrical sector composite surface
+
+    This class acts as a proper surface, meaning that unary `+` and `-`
+    operators applied to it will produce a half-space. The negative
+    side is defined to be the region inside of the cylinder sector.
+
+    Parameters
+    ----------
+    center : iterable of float
+       Coordinate for central axes of cylinders in the (y, z), (z, x), or
+         (x, y) basis. Defaults to (0,0)
+    r1 : float
+        Inner cylinder radii
+    r2 : float
+        Outer cylinder radii
+    theta0 : float
+        Angular offset of the sector in degrees relative to the primary basis
+        axis (+y, +z, or +x).
+    theta : float
+        Angular width of the sector in degrees.
```

Comment:
To prevent confusion, I think we should have the user specify two angles theta1 and theta2 rather than an angle and an angular width.

Reply:
I think having an offset and a width is a bit more robust of an approach than two explicit angles, but I agree that having two `theta` variables is confusing... maybe `delta` would be a good variable for the angular width?

I actually initially thought about using a two-angle approach like you suggested, but one of the issues I ran into was handling cases where the two angles are inverted, i.e. `theta1 = 30` and `theta2 = 40` would be a 10 degree sector, whereas `theta1 = 40` and `theta2 = 30`, would be a 350 degree sector, but this isn't entirely clear from just reading off the values of `theta1` and `theta2`. The offset and sector width formulation is clearer about this difference.

One snag I can see right now though is it's not entirely clear where on the sector the offset is measured to. In my head it's the clockwise-most edge of the sector but this isn't specified in the docstring.

Reply:
One way to get rid of any ambiguities with two angles is to require that theta2 > theta1

Reply:
I've changed the variable names to be more specific. Instead of `theta` we have `central_angle`, and instead of `theta0` we have `ccw_offset`. I've also changes the internal variables `theta0` and `theta1` to `phi1` and `phi2`, respectively.